addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -255,7 +255,7 @@ def format_alphanumeric(text_to_convert):
 
         # Reduce downpayment views to a single recordset
         downpayment_moves = [l.get('downpayment_moves', self.env['account.move']) for l in invoice_lines]
-        downpayment_moves = reduce(lambda x, y: x | y, downpayment_moves)
+        downpayment_moves = reduce(lambda x, y: x | y, downpayment_moves) if downpayment_moves else []
```

Comment:
From O(n^2) to O(n)
```suggestion
        downpayment_moves = self.browse(move.id for moves in downpayment_moves for move in moves)
```

Could even be merged with the line above for more readability

Reply:
Hello @william-andre ,
Thank you for the suggestion. I have done the changes. Please review them once.